backend/api/events/sync.py

```diff
@@ -382,7 +383,17 @@ def _disable_model(key: str) -> None:
 def _on_create_failure_report(event: dict) -> None:
     """Process create failure report event to update local database."""
     logger.debug("Syncing failure report create", asset_key=event["asset_key"], event_id=event["id"])
-    _update_computetask(key=event["asset_key"], failure_report=event["failure_report"])
+
+    asset_key = event["asset_key"]
+    failure_report = event["failure_report"]
+    asset_type = failure_report_pb2.FailedAssetKind.Value(failure_report["asset_type"])
+
+    if asset_type == failure_report_pb2.FAILED_ASSET_FUNCTION:
+        # Needed as this field is only in ComputeTask
+        compute_task_key = ComputeTask.objects.values_list("key", flat=True).get(function_id=asset_key)
```

Comment:
I think we can have [`Model.MultipleObjectsReturned`](https://docs.djangoproject.com/en/4.2/ref/models/class/#django.db.models.Model.MultipleObjectsReturned) exception here, because several compute tasks can be linked to the same function.

Reply:
We probably should use rather iterate on the queryset (ensuring that it's not empty) to update all relevant CT status

Reply:
Filtering on status STATUS_DOING & STATUS_TODO